### PR TITLE
feat(stream): backfill resuming for sink

### PR DESCRIPTION
**Pull request by kwannoel**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Closes https://github.com/risingwavelabs/risingwave/issues/12080.

- Recovery of the backfill progress `(rw_ddl_progress)` is unsupported currently. I will work on that separately.
- The backfill progress tracking is still visible if cluster did not restart.
- This PR makes sure that sink is recoverable even if backfill fails.
- This PR makes sure sink is immediately visible after first barrier, independent of whether backfill is complete.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

`SINK` DDL statements no longer need to wait for backfill to complete.

</details>

### CODE REVIEW

**Review by chenzl25 — APPROVED** [review]
LGTM! Maybe we should add a user-facing tag.

**Review by yezizp2012 — APPROVED** [review]
LGTM

File: src/meta/src/barrier/progress.rs [hunk 1]
```diff
@@ -153,6 +153,8 @@ impl Progress {
 ///    On recovery, the stream manager will stop managing the job.
 /// 2. `Recovered`. This refers to the "Recovered" type of tracking job.
 ///    On recovery, the barrier manager will recover and start managing the job.
+/// 3. `Immediate`. Immediate jobs will immediately return,
+///    but still register the lag between the job and its upstream.
```
Comment (BugenZhao) [inline src/meta/src/barrier/progress.rs:157]: Didn't find this variant? 👀

Comment (kwannoel) [reply src/meta/src/barrier/progress.rs:157]: Removed in https://github.com/risingwavelabs/risingwave/pull/13735/files, thanks!

It was used previously to indicate a new variant of stream job handling, but I realize I can just use a TrackingCommand without notifiers to accomplish the same thing, with less refactoring.

**Pull request merged**